xdsl/parser/generic_parser.py

```diff
@@ -307,9 +307,25 @@ def parse_characters(self, text: str, context_msg: str = "") -> str:
         self.raise_error(f"'{text}' expected" + context_msg)
 
     @contextmanager
-    def in_angle_brackets(self):
-        self.parse_characters("<")
-        try:
+    def delimited(self, delimiter: Delimiter):
```

Comment:
why not re-use the API of printer?
```suggestion
    def delimited(self, start: str, end: str):
```

Reply:
I could do, kind of makes me wonder what the point of `Parser.Delimiter` was

Reply:
I might be missing something